File: examples/counter/src/main/java/com/brianegan/bansa/counter/StoreModule.java
package com.brianegan.bansa.counter;

import com.brianegan.bansa.Action;
import com.brianegan.bansa.CreateStoreKt;
import com.brianegan.bansa.Store;
import dagger.Module;
import dagger.Provides;
import rx.schedulers.Schedulers;

import javax.inject.Singleton;

@Module(library = true)
public class StoreModule {
    public StoreModule() {
    }

    @Provides
    @Singleton
    public Store<ApplicationState, Action> provideStore() {
        return CreateStoreKt.createStore(
                new ApplicationState(),
                ApplicationReducerKt.getApplicationReducer(),
                Schedulers.newThread()
        );
    }
}

File: examples/counter/src/main/kotlin/com/brianegan/bansa/counter/StoreModule.kt
package com.brianegan.bansa.counter

import com.brianegan.bansa.Store
import com.brianegan.bansa.createStore
import uy.kohesive.injekt.api.InjektModule
import uy.kohesive.injekt.api.InjektRegistrar
import uy.kohesive.injekt.api.addSingleton

class StoreModule : InjektModule {
    override fun InjektRegistrar.registerInjectables() {
        addSingleton<Store<ApplicationState, Any>>(createStore(ApplicationState(), counterReducer))
    }
}
